lib/Doctrine/ORM/Mapping/Builder/AssociationBuilder.php

```diff
@@ -58,7 +58,7 @@ public function __construct(ClassMetadataBuilder $builder, array $mapping, $type
     /**
      * @param string $fieldName
      *
-     * @return AssociationBuilder
+     * @return self
```

Comment:
`static` as return type everywhere in this file, since it's inherited by more specialized AssociationBuilders